### DISCUSSION TITLE
Incapable to build Snap package

### DISCUSSION THREAD

**Discussion by mhalano** [fwupd/fwupd#6620] [Q&A]
Describe the bug
There is some traces of information to build a snap package, but it doesn't work end to end.

Steps to Reproduce
Execute:
OS=snap-x86_64 ./contrib/ci//generate_docker.py build
docker run --privileged -t -v pwd:/github/workspace fwupd-snap-x86_64

Expected behavior
I was expecting a .snap file to be generated in my `pwd` directory.

fwupd version information
Commit https://github.com/fwupd/fwupd/commit/c4f8ef408b6cee2cfd4b90929b99e3677dc0f8a3

Additional questions

    Operating system and version: Ubuntu 23.10
    Have you tried rebooting? No
    Is this a regression? Maybe. I don't know if the Snap package generation worked someday.

**Comment by superm1** [comment]
Run `snapcraft` and it will build a snap.

**Comment by mhalano** [comment]
My idea would be to be able to use the same logic with the others package formats to create a snap package. Also I got this error when I executed the snapcraft command, even there is the file:
```
/snap/bin/snapcraft: 12: exec: /snap/snapcraft/current/usr/bin/python3: not found
```
@superm1 What would be the full sequence of commands to generate a .snap package correctly?

**Reply by superm1** [reply]
I'm not sure why you're hitting that.  Maybe you're missing a core snap?  Here is how we build the snap:
https://github.com/fwupd/fwupd/blob/main/.github/workflows/snap.yml

I've locally just run `snapcraft` and it works for me, but I might have something else installed that you don't?

**Reply by superm1** [reply]
Maybe you need to make the symlink that the job makes?
